From 9ced0261192be0072b60de79b894bc59ccef77c9 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@digia.com>
Date: Wed, 22 Jan 2014 15:49:39 +0100
Subject: [PATCH 20/20] Make it possible to build without using build-webkit
 script

This makes qmake 4 also evaluate features.prf, and caches the result
in .qmake.cache.
---
 .qmake.cache.in    |  2 ++
 Makefile           | 17 -----------------
 Source/WTF/WTF.pri |  2 ++
 WebKit.pro         |  3 +++
 4 files changed, 7 insertions(+), 17 deletions(-)
 create mode 100644 .qmake.cache.in
 delete mode 100644 Makefile

diff --git a/.qmake.cache.in b/.qmake.cache.in
new file mode 100644
index 0000000..d8ed6e8
--- /dev/null
+++ b/.qmake.cache.in
@@ -0,0 +1,2 @@
+DEFINES = $$DEFINES
+CONFIG = $$CONFIG
\ No newline at end of file
diff --git a/Makefile b/Makefile
deleted file mode 100644
index 19d889a..0000000
--- a/Makefile
+++ /dev/null
@@ -1,17 +0,0 @@
-MODULES = Source Tools
-
-all:
-	@for dir in $(MODULES); do ${MAKE} $@ -C $$dir; exit_status=$$?; \
-	if [ $$exit_status -ne 0 ]; then exit $$exit_status; fi; done
-
-debug d development dev develop:
-	@for dir in $(MODULES); do ${MAKE} $@ -C $$dir; exit_status=$$?; \
-	if [ $$exit_status -ne 0 ]; then exit $$exit_status; fi; done
-
-release r deployment dep deploy:
-	@for dir in $(MODULES); do ${MAKE} $@ -C $$dir; exit_status=$$?; \
-	if [ $$exit_status -ne 0 ]; then exit $$exit_status; fi; done
-
-clean:
-	@for dir in $(MODULES); do ${MAKE} $@ -C $$dir; exit_status=$$?; \
-	if [ $$exit_status -ne 0 ]; then exit $$exit_status; fi; done
diff --git a/Source/WTF/WTF.pri b/Source/WTF/WTF.pri
index 9423b53..f1ce0a8 100644
--- a/Source/WTF/WTF.pri
+++ b/Source/WTF/WTF.pri
@@ -22,6 +22,8 @@ haveQt(5) {
             error("To build QtWebKit with Qt 5 you need to build Qt 5 with libICU support. Check for ICU support being mentioned in qtbase/config.summary.")
         }
     }
+} else {
+    load(.qmake.cache)
 }
 
 linux-*:contains(DEFINES, WTF_USE_GSTREAMER=1) {
diff --git a/WebKit.pro b/WebKit.pro
index 75b9159..fcc7c8d 100644
--- a/WebKit.pro
+++ b/WebKit.pro
@@ -11,6 +11,9 @@ equals(QT_MAJOR_VERSION, 4) {
         error("The environment variable QMAKEPATH needs to point to $WEBKITSRC/Tools/qmake")
         # Otherwise we won't pick up the feature prf files needed for the build
     }
+    CONFIG += production_build
+    include(Tools/qmake/configure.pri)
+    QMAKE_SUBSTITUTES += .qmake.cache.in
 } else:!webkit_configured {
     CONFIG += webkit_configured
     CONFIG += production_build
-- 
1.8.5.3

